comms/storage/contentaccess/content_access_middleware.go

```diff
@@ -0,0 +1,78 @@
+package contentaccess
+
+import (
+	"encoding/json"
+	"net/url"
+	"strconv"
+
+	"github.com/labstack/echo/v4"
+)
+
+func ContentAccessMiddleware(next echo.HandlerFunc) echo.HandlerFunc {
+	return func(c echo.Context) error {
+
+		requestedCid := c.Param("CID")
+		if requestedCid == "" {
+			return echo.ErrBadRequest
+		}
+
+		signatureData, signature, err := parseQueryParams(c.QueryParams())
+		if err != nil {
+			return echo.ErrBadRequest
+		}
+
+		isValidSignature, err := VerifySignature(*signatureData, []byte(signature), requestedCid)
+		if err != nil {
+			return echo.ErrBadGateway
+		}
+
+		if !isValidSignature {
+			return echo.ErrBadRequest
+		}
+
+		c.Response().Header().Set("shouldCache", strconv.FormatBool(signatureData.ShouldCache))
```

Comment:
This is different to how it's done on content nodes so I'd really like feedback on it

Reply:
glad you asked because I would've glossed over it otherwise. `shouldCache` isn't a valid response header - I think you should set `shouldCache` on the request instead of the response. it looks like content node sets a `req.shouldCache` flag here in contentAccessMiddleware and then later when streaming [here](https://github.com/AudiusProject/audius-protocol/blob/main/creator-node/src/routes/files.js#L126-L134) uses that flag to set the `cache-control` header to either 30 days or no-cache if it's premium content

Reply:
Ya I was trying to avoid setting it on the request but it makes sense why it can't be a response header lol. I'll see if I can avoid creating a custom Request struct but it push comes to shove then that's what I'll do